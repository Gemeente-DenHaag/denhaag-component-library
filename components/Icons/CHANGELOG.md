# @gemeente-denhaag/icons

## 1.0.0

### Major Changes

- 281fa34: Added wallet icon
