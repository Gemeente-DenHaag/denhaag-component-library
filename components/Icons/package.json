{
  "name": "@gemeente-denhaag/icons",
  "version": "0.2.3-alpha.84",
  "description": "The icon set of the Municipality of The Hague",
  "bugs": "https://github.com/nl-design-system/denhaag/issues",
  "repository": {
    "type": "git",
    "url": "https://github.com/nl-design-system/denhaag.git",
    "directory": "components/Icons"
  },
  "license": "EUPL-1.2",
  "author": "Municipality of The Hague",
  "exports": "./dist/index.js",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c ../../rollup.config.js",
    "clean": "yarn rimraf dist tsconfig.tsbuildinfo"
  },
  "dependencies": {
    "@gemeente-denhaag/baseprops": "0.2.3-alpha.84",
    "@material-ui/core": "4.12.3"
  },
  "peerDependencies": {
    "react": "^17.0.0"
  }
}
