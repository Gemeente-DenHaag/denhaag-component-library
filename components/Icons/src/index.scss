.denhaag-icon {
  aspect-ratio: 1;
  display: inline-block;
  fill: currentcolor;
  flex-shrink: 0;
  font-size: var(--denhaag-icon-size, var(--denhaag-space-xl));
  height: 1em;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  width: 1em;
}
