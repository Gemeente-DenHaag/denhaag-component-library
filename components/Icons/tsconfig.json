{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "dist",
    "typeRoots": ["./src/@types"],
    "composite": true
  },
  "include": ["src/**/*"],
  "exclude": ["dist", "src/**/*.stories.tsx"]
}
