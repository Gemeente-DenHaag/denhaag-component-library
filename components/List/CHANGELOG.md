# @gemeente-denhaag/list

## 0.2.3

### Patch Changes

- Updated dependencies [281fa34]
  - @gemeente-denhaag/icons@1.0.0
  - @gemeente-denhaag/iconbutton@0.2.3-alpha.422
