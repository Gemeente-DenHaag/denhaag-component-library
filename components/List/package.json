{
  "name": "@gemeente-denhaag/list",
  "version": "0.2.3-alpha.145",
  "description": "A List component",
  "bugs": "https://github.com/nl-design-system/denhaag/issues",
  "repository": {
    "type": "git",
    "url": "https://github.com/nl-design-system/denhaag.git",
    "directory": "components/List"
  },
  "license": "EUPL-1.2",
  "author": "Municipality of The Hague",
  "exports": "./dist/index.js",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c ../../rollup.config.js",
    "clean": "yarn rimraf dist tsconfig.tsbuildinfo"
  },
  "dependencies": {
    "@gemeente-denhaag/iconbutton": "0.2.3-alpha.145",
    "@gemeente-denhaag/icons": "0.2.3-alpha.145"
  },
  "peerDependencies": {
    "react": "^17.0.0"
  },
  "gitHead": "dcf72a9b79266c1ebede35aff4a02dd9121a980f"
}
