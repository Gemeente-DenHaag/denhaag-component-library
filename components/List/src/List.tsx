import React from 'react';
import { List as MaterialList } from '@material-ui/core';
import BaseProps from '@gemeente-denhaag/baseprops';
import { ListSubheaderProps } from './ListSubheader';
import { list_classes as classes } from './bem-mapping';

import './mui-override.css';
import './listitem.css';
import './listitemsecondaryaction.css';
import './listitemicon.css';
import './listitemtext.css';
import './listsubheader.css';

export interface ListProps extends Omit<BaseProps, 'classes'> {
  /**
   * The content of the subheader, normally `ListSubheader`.
   */
  subheader: React.ReactElement<ListSubheaderProps>;
}

/**
 * Primary UI component for user interaction
 */
export const List: React.FC<ListProps> = (props: ListProps) => {
  const materialProps = { subheader: props.subheader };

  return (
    <MaterialList {...materialProps} classes={classes}>
      {props.children}
    </MaterialList>
  );
};

export default List;
