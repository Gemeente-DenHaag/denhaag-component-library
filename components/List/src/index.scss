@import "./list-item-icon";
@import "./list-item-secondary-action";
@import "./list-item-text";
@import "./list-item";
@import "./list-subheader";
@import "./list";
