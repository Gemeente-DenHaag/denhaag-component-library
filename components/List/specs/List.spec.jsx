import * as React from 'react';
import List, { ListItem, ListSubheader } from '../src';
import { HouseIcon, TrashIcon } from '../../Icons';

describe(`List tests`, () => {
  it('has clickable listitems', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('List item click');

    cy.mount(
      <List>
        <ListItem
          primaryText="Primary text"
          secondaryText="Secondary text"
          actionType="nav"
          onClick={props.onClick}
          rightIcon={<TrashIcon />}
        />
      </List>,
    );

    cy.get('.denhaag-list__item').contains('Primary text');
    cy.get('.denhaag-list__item').contains('Secondary text');

    cy.get('.denhaag-list__item')
      .click()
      .then(() => expect(spy).to.have.been.calledOnce);

    // onClick called when action button is clicked
    cy.get('.denhaag-list__item .denhaag-list__secondary-action button')
      .click()
      .then(() => expect(spy).to.have.been.calledTwice);
  });

  it('has clickable secondary action on actionType action', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('List item action click');

    cy.mount(
      <List>
        <ListItem
          primaryText="Primary text"
          secondaryText="Secondary text"
          actionType="action"
          onClick={props.onClick}
          leftIcon={<HouseIcon color="inherit"></HouseIcon>}
          rightIcon={<TrashIcon />}
        />
      </List>,
    );

    // onClick called when action button is clicked
    cy.get('.denhaag-list__item .denhaag-list__secondary-action button')
      .click()
      .then(() => expect(spy).to.have.been.calledOnce);

    // onClick not called when whole list item is clicked
    cy.get('.denhaag-list__item')
      .click()
      .then(() => expect(spy).to.have.been.calledOnce);
  });

  it('has proper keyboard support for actionType nav', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('List item enter');

    cy.mount(
      <List>
        <ListItem
          primaryText="Primary text"
          secondaryText="Secondary text"
          actionType="nav"
          onClick={props.onClick}
          rightIcon={<TrashIcon />}
        />
      </List>,
    );

    cy.get('.denhaag-list__item')
      .focus()
      .type('{enter}')
      .then(() => expect(spy).to.have.been.calledOnce);

    // Secondary action should have tabindex -1
    cy.get('.denhaag-list__item .denhaag-list__secondary-action button').should('have.attr', 'tabindex', '-1');

    // List item should have tabindex 0
    cy.get('.denhaag-list__item').should('have.attr', 'tabindex', '0');
  });

  it('has proper keyboard support for actionType action', () => {
    const props = {
      onClick: () => {
        console.log('Clicked');
      },
    };

    const spy = cy.spy(props, 'onClick').as('List item action enter');

    cy.mount(
      <List>
        <ListItem
          primaryText="Primary text"
          secondaryText="Secondary text"
          actionType="action"
          onClick={props.onClick}
          rightIcon={<TrashIcon />}
        />
      </List>,
    );

    cy.get('.denhaag-list__item .denhaag-list__secondary-action button')
      .type('{enter}')
      .then(() => expect(spy).to.have.been.calledOnce);

    cy.get('.denhaag-list__item')
      .focus()
      .type('{enter}')
      .then(() => expect(spy).to.have.been.calledOnce);

    // Secondary action should have tabindex 0
    cy.get('.denhaag-list__item .denhaag-list__secondary-action button').should('have.attr', 'tabindex', '0');

    // List item should have tabindex -1
    cy.get('.denhaag-list__item').should('have.attr', 'tabindex', '-1');
  });

  it('a11y and snapshots', () => {
    cy.snapshots(
      <List subheader={<ListSubheader>subheader</ListSubheader>}>
        <ListItem primaryText="ListItem" />
        <ListItem
          primaryText="Single-line item"
          actionType="action"
          onClick={console.log('TrashIcon click')}
          leftIcon={<HouseIcon color="inherit"></HouseIcon>}
          rightIcon={<TrashIcon />}
        />
        <ListItem
          primaryText="Single-line item"
          secondaryText="Secondary text"
          onClick={console.log('ListItem click')}
          leftIcon={<HouseIcon color="inherit" />}
          actionType="nav"
        />
      </List>,
      {},
      {
        selector: '.denhaag-list__item',
        states: {
          hover: 'denhaag-list__item--hover',
          focus: 'denhaag-list__item--focus',
        },
      },
      [{ id: 'list', issue: 402 }],
    );
  });
});
