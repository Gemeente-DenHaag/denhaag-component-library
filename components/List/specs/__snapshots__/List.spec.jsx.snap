exports[`List tests > a11y and snapshots #0`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <ul class="MuiList-root denhaag-list MuiList-padding MuiList-subheader">
      <p class="MuiTypography-root denhaag-list__subheader MuiTypography-body1">
        subheader
      </p>
      <li
        class="MuiListItem-root denhaag-list__item MuiListItem-gutters"
        primarytext="ListItem"
        tabindex="0"
      >
        <div class="MuiListItemText-root denhaag-list__item-text">
          <p
            class="MuiTypography-root denhaag-list__item-text-primary MuiTypography-body1"
          >
            ListItem
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root denhaag-list__item MuiListItem-gutters"
        actiontype="action"
        primarytext="Single-line item"
        tabindex="-1"
      >
        <div class="MuiListItemIcon-root denhaag-list__item-icon">
          <svg
            width="1em"
            height="1em"
            viewBox="0 0 24 24"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
            class="MuiSvgIcon-root denhaag-icon"
            focusable="false"
            aria-hidden="true"
            shape-rendering="auto"
          >
            <path
              d="M11.336 2.253a1 1 0 011.328 0l9 8a1 1 0 01-1.328 1.494L20 11.45V19a2 2 0 01-2 2H6a2 2 0 01-2-2v-7.55l-.336.297a1 1 0 01-1.328-1.494l9-8zM6 9.67V19h3v-5a1 1 0 011-1h4a1 1 0 011 1v5h3V9.671l-6-5.333-6 5.333zM13 19v-4h-2v4h2z"
              fill="currentColor"
            ></path>
          </svg>
        </div>
        <div
          class="MuiListItemSecondaryAction-root denhaag-list__secondary-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root denhaag-icon-button"
            tabindex="0"
            type="button"
            aria-label="Action"
          >
            <span class="MuiIconButton-label"
              ><svg
                width="1em"
                height="1em"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
                class="MuiSvgIcon-root denhaag-icon"
                focusable="false"
                aria-hidden="true"
                shape-rendering="auto"
              >
                <path
                  d="M7 4a2 2 0 012-2h6a2 2 0 012 2v2h4a1 1 0 110 2h-1.069l-.867 12.142A2 2 0 0117.069 22H6.93a2 2 0 01-1.995-1.858L4.07 8H3a1 1 0 010-2h4V4zm2 2h6V4H9v2zM6.074 8l.857 12H17.07l.857-12H6.074zM10 10a1 1 0 011 1v6a1 1 0 11-2 0v-6a1 1 0 011-1zm4 0a1 1 0 011 1v6a1 1 0 11-2 0v-6a1 1 0 011-1z"
                  fill="currentColor"
                ></path></svg
            ></span>
          </button>
        </div>
        <div class="MuiListItemText-root denhaag-list__item-text">
          <p
            class="MuiTypography-root denhaag-list__item-text-primary MuiTypography-body1"
          >
            Single-line item
          </p>
        </div>
      </li>
      <li
        class="MuiButtonBase-root MuiListItem-root denhaag-list__item MuiListItem-gutters MuiListItem-button"
        tabindex="0"
        aria-disabled="false"
        actiontype="nav"
        secondarytext="Secondary text"
        primarytext="Single-line item"
      >
        <div class="MuiListItemIcon-root denhaag-list__item-icon">
          <svg
            width="1em"
            height="1em"
            viewBox="0 0 24 24"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
            class="MuiSvgIcon-root denhaag-icon"
            focusable="false"
            aria-hidden="true"
            shape-rendering="auto"
          >
            <path
              d="M11.336 2.253a1 1 0 011.328 0l9 8a1 1 0 01-1.328 1.494L20 11.45V19a2 2 0 01-2 2H6a2 2 0 01-2-2v-7.55l-.336.297a1 1 0 01-1.328-1.494l9-8zM6 9.67V19h3v-5a1 1 0 011-1h4a1 1 0 011 1v5h3V9.671l-6-5.333-6 5.333zM13 19v-4h-2v4h2z"
              fill="currentColor"
            ></path>
          </svg>
        </div>
        <div
          class="MuiListItemSecondaryAction-root denhaag-list__secondary-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root denhaag-icon-button"
            tabindex="-1"
            type="button"
            aria-label="Navigate"
          >
            <span class="MuiIconButton-label"
              ><svg
                width="1em"
                height="1em"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
                class="MuiSvgIcon-root denhaag-icon"
                focusable="false"
                aria-hidden="true"
                shape-rendering="auto"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                ></path></svg
            ></span>
          </button>
        </div>
        <div
          class="MuiListItemText-root denhaag-list__item-text MuiListItemText-multiline"
        >
          <p
            class="MuiTypography-root denhaag-list__item-text-primary MuiTypography-body1"
          >
            Single-line item
          </p>
          <p
            class="MuiTypography-root denhaag-list__item-text-secondary MuiTypography-body1"
          >
            Secondary text
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
